File: C++/Maths/Find CGPA Letter Grade by obtained marks.cpp
#include <iostream>
using namespace std;

int main(){
    int marks;
    cout << "Enter the obtained marks out of 100: ";
    cin >> marks;

    string grade; double points;
    if(marks > 100){
        grade = "invalid";
    }else if(marks >= 80){
        grade = "A+";
    }else if(marks >= 75){
        grade = "A";
    }else if(marks >= 70){
        grade = "A-"; 
    }else if(marks >= 65){
        grade = "B+";
    }else if(marks >= 60){
        grade = "B";
    }else if(marks >= 55){
        grade = "B-";
    }else if(marks >= 50){
        grade = "C+";
    }else if(marks >= 45){
        grade = "C";
    }else if(marks >= 40){
        grade = "D";
    }else{
        grade = "F";
    }

    if(grade == "invalid"){
        printf("\nInvalid marks. Result not found.");
    }else if(points == 0.00){
        printf("\nResult Failed \nGrade: %s", grade.c_str());
    }else{
        printf("\nResult Passed \nGrade: %s", grade.c_str());
    }
}

File: Python/Maths/Find CGPA Letter Grade by obtained marks.py
marks = int(input("Enter the obtained marks out of 100: "))
if marks > 100:
    grade = "invalid"
elif marks >= 80:
    grade = "A+"
elif marks >= 75:
    grade = "A"
elif marks >= 70:
    grade = "A-"
elif marks >= 65:
    grade = "B+"
elif marks >= 60:
    grade = "B"
elif marks >= 55:
    grade = "B-"
elif marks >= 50:
    grade = "C+"
elif marks >= 45:
    grade = "C"
elif marks >= 40:
    grade = "D"
else:
    grade = "F"
    
# Advance method used for more validation
if grade == "invalid":
    print("\nInvalid marks. Result not found.")
elif grade == 0.00:
    print("\nResult Failed \nGrade:", grade)
else:
    print("\nResult Passed \nGrade:", grade)